services/app-web/src/common-code/data-helpers/makeDocumentKeyLookupList.ts

```diff
@@ -0,0 +1,52 @@
+import { base64ToDomain } from '../proto/stateSubmission'
+import { HealthPlanPackage } from '../../gen/gqlClient'
+import { parseKey } from '../s3URLEncoding'
+
+type LookupListType = {
+    currentDocuments: string[]
+    previousDocuments: string[]
+}
+
+const getKey = (s3URL: string) => {
+    const key = parseKey(s3URL)
+    return key instanceof Error ? null : key
+}
+
+export const makeDocumentList = (
+    submissions: HealthPlanPackage
+): LookupListType | undefined => {
+    const docBuckets = [
+        'contractDocuments',
+        'rateDocuments',
+        'documents',
+    ] as const
+    const lookupList: LookupListType = {
+        currentDocuments: [],
+        previousDocuments: [],
+    }
+    if (submissions) {
+        const revisions = submissions.revisions
+        if (revisions.length > 1) {
```

Comment:
I’m curious about this if, too, there’s no harm in calling .forEach() on an empty array. 

In general I’ve been prefering the `for (const revision of revisions) {}` syntax, but I don’t know if you can get an index out of that. I think it might be a little more elegant to make a separate function that takes a revision and an array to dump its documents into and call that differently for the first and then the rest of the reivisions.

Reply:
I don't believe you can get an index using `for..of` by itself. Usually you can do something like 
`for ( const [index, revision] of revisions.entries()) {...}`
but typescrypt does't like `revisions.entries()`

I could use  the standard `for` loop with an `iterator` or map the revision to an index like:
`const mappedRevisions = revisions.map((revision, index)=> ({ index, revision}))` then you could for `for...of` but that doesn't seem efficient.